### ISSUE TITLE
MessageID always seems to serialize to empty slice in Go client Reader

### ISSUE THREAD

**Issue by tobgu** [apache/pulsar#3177]
#### Expected behavior

```go
// Error checks left for brevity, these are checked in real code
reader, _ := client.CreateReader(pulsar.ReaderOptions{
	Topic:          "foo",
	StartMessageID: pulsar.LatestMessage})

ctx, cancel := context.WithTimeout(context.Background(), 1000 * time.Millisecond)
defer cancel()
msg, err := r.reader.Next(ctx)
if err != nil {
   // I would expect this to print a byte slice with length > 0 but it's
   // always an empty slice...
   fmt.Printf("ID: %v", msg.ID().Serialize())
}
```
#### Actual behavior

Message ID always seems to be empty.

#### Steps to reproduce
See example above. The behaviour is the same on a standalone cluster and a full blown cluster with 3 x zookeeper, 3 x broker and 3 x bookie.

#### System configuration
**Pulsar version**: 2.2.0
**Go client version**:
github.com/apache/pulsar v2.2.0+incompatible h1:i2dfg67DNdHRgdzdIiSObTb9VRavAtZStxC4YIySxAI=
github.com/apache/pulsar v2.2.0+incompatible/go.mod h1:7npmYEEu+jEeIMdpGSJwrTxUB86OW/UM5IkJCAyfl5Y=
g

**Comment by Tevic** [comment]
I've figure out why and test success.
`
https://github.com/apache/pulsar/pull/3801
`
But if use DeserializeMessageID and use consumer.AckID it will panic, i don't know what cause the panic.
`
fatal error: unexpected signal during runtime execution
[signal SIGSEGV: segmentation violation code=0x1 addr=0x8 pc=0x7fdbd4e1127f]

runtime stack:
runtime.throw(0x517df2, 0x2a)
        /home/tevic/Work/RunTime/Go/src/runtime/panic.go:617 +0x72
runtime.sigpanic()
        /home/tevic/Work/RunTime/Go/src/runtime/signal_unix.go:374 +0x4a9

goroutine 1 [syscall]:
runtime.cgocall(0x4cceb0, 0xc0000dfbb0, 0x7e71c0)
        /home/tevic/Work/RunTime/Go/src/runtime/cgocall.go:128 +0x5b fp=0xc0000dfb80 sp=0xc0000dfb48 pc=0x4096bb
github.com/apache/pulsar/pulsar-client-go/pulsar._Cfunc_pulsar_consumer_acknowledge_async_id(0x7fdba800e8e0, 0x1e61710, 0x0, 0x0)
        _cgo_gotypes.go:761 +0x45 fp=0xc0000dfbb0 sp=0xc0000dfb80 pc=0x4b83a5
github.com/apache/pulsar/pulsar-client-go/pulsar.(*consumer).AckID.func1(0xc00009c080, 0x52ff40, 0xc0000a8058)
        /home/tevic/Work/WorkSpace/Code/Go/src/github.com/apache/pulsar/pulsar-client-go/pulsar/c_consumer.go:243 +0xc2 fp=0xc0000dfbf8 sp=0xc0000dfbb0 pc=0x4c3c72
github.com/apache/pulsar/pulsar-client-go/pulsar.(*consumer).AckID(0xc00009c080, 0x52ff40, 0xc0000a8058, 0x1, 0x1)
        /home/tevic/Work/WorkSpace/Code/Go/src/github.com/apache/pulsar/pulsar-client-go/pulsar/c_consumer.go:243 +0x3f fp=0xc0000dfc20 sp=0xc0000dfbf8 pc=0x4bddcf
main.main()
        /home/tevic/Work/WorkSpace/Code/GoProjs/HelloWorld/Pulsar/Recv.go:46 +0x7f3 fp=0xc0000dff98 sp=0xc0000dfc20 pc=0x4cc063
runtime.main()
        /home/tevic/Work/RunTime/Go/src/runtime/proc.go:200 +0x20c fp=0xc0000dffe0 sp=0xc0000dff98 pc=0x4319cc
runtime.goexit()
        /home/tevic/Work/RunTime/Go/src/runtime/asm_amd64.s:1337 +0x1 fp=0xc0000dffe8 sp=0xc0000dffe0 pc=0x45aa21

goroutine 17 [chan send, locked to thread]:
github.com/apache/pulsar/pulsar-client-go/pulsar.pulsarMessageListenerProxy(0x7fdbc1ffa7e0, 0x7fdba0000d00, 0x1e636c0)
        /home/tevic/Work/WorkSpace/Code/Go/src/github.com/apache/pulsar/pulsar-client-go/pulsar/c_consumer.go:192 +0x11b
github.com/apache/pulsar/pulsar-client-go/pulsar._cgoexpwrap_40dd75c63fd4_pulsarMessageListenerProxy(0x7fdbc1ffa7e0, 0x7fdba0000d00, 0x1e636c0)
        _cgo_gotypes.go:1867 +0x3f
exit status 2
`

**Comment by sijie** [comment]
This is fixed by #3801
